### PR TITLE
Make defining a port number for hosts in a cluster that only connect …

### PR DESCRIPTION
**Pull request by sethhall**
…outbound optional

### CODE REVIEW

File: scripts/base/frameworks/cluster/main.zeek [hunk 1]
```diff
@@ -1,4 +1,4 @@
-##! A framework for establishing and controlling a cluster of Zeek instances.
+#hh#! A framework for establishing and controlling a cluster of Zeek instances.
```
Comment (jsiwek) [inline scripts/base/frameworks/cluster/main.zeek:1]:
```suggestion
##! A framework for establishing and controlling a cluster of Zeek instances.
```

File: scripts/base/frameworks/cluster/main.zeek [hunk 2]
```diff
@@ -162,7 +162,7 @@ export {
 		## can specify a particular :rfc:`4007` ``zone_id``.
 		zone_id:      string      &default="";
 		## The port that this node will listen on for peer connections.
-		p:            port;
+		p:            port        &optional;
```
Comment (jsiwek) [inline scripts/base/frameworks/cluster/main.zeek:165]:
```suggestion
		## A value of ``0/unknown`` means the node is not pre-configured to listen.
		p:            port        &default=0/unknown;
```

File: scripts/base/frameworks/cluster/setup-connections.zeek [hunk 3]
```diff
@@ -89,11 +89,15 @@ event zeek_init() &priority=-10
 	Broker::subscribe(nodeid_topic(Broker::node_id()));
 	Broker::subscribe(node_topic(node));
 
-	Broker::listen(Broker::default_listen_address,
-	               self$p,
-	               Broker::default_listen_retry);
+	if ( self?$p )
```
Comment (jsiwek) [inline scripts/base/frameworks/cluster/setup-connections.zeek:92]:
```suggestion
	if ( self$p != 0/unknown )
```

**Review by jsiwek — COMMENTED** [review]
Using a `&default` instead of `&optional` may be a bit more backwards-compatible (may still be worth a note in NEWS).  e.g. if there's a script out there that was iterating nodes and printing specific fields, the sudden change to `&optional` would start causing runtime errors and be "more broken" than if it just starts printing out the new default/sentinel value.  What do you think?

**Comment by sethhall** [comment]
I think your suggestions made sense so I incorporated them.

**Pull request merged**